doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -668,69 +652,83 @@ Induction
+      See https://github.com/coq/coq/pull/14179#discussion_r642193096
+
+   .. todo PR: following is clear as mud
 
    Let the type of :n:`@ident` in the local context be :g:`(I t)`, where :g:`I`
-   is a (co)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
+   is a (co-)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
    derives for each possible constructor :g:`c i` of :g:`(I t)`, all the
    necessary conditions that should hold for the instance :g:`(I t)` to be
    proved by :g:`c i`.
 
+   :n:`@ident`
+     The name of the hypothesis to apply.
+
+   :n:`@natural`
+     Equivalent to :n:`intros until @natural; inversion @ident`
+     where :n:`@ident` is the identifier for the last introduced hypothesis.
+
+   :n:`{? in {+ @ident } }`
+     When :n:`{+ @ident}` are identifiers in the local context, this causes
+     generalizing on :n:`{+ @ident}` as the initial step of `inversion`.
+     See :tacn:`generalize`.
+
+   .. todo PR: "generalizing"?  As in :n:`generalize @ident__1; ... generalize @ident__n` ?
+
+   :n:`as @or_and_intropattern`
+      Provides names for the variables introduced in each new subgoal.  The
+      :token:`or_and_intropattern` must have one :n:`{* @intropattern }`
+      for each constructor of the type of :n:`(I t)`, given in the order
+      in which the constructors are defined.
+      If there are not enough names, Coq picks fresh names.
+
+     If an equation splits into several
+     equations (because ``inversion`` applies ``injection`` on the equalities it
+     generates), the corresponding :n:`@intropattern` should be in the form of
+     :n:`@or_and_intropattern` (or the equivalent :n:`{*, ( @simple_intropattern ) }`),
+     with the number of entries equal to the number
+     of subequalities obtained from splitting the original equation.
+
+     Here is an example: the ``inversion … as`` variant of
+     ``inversion`` generally behaves in a slightly more expected way than
+     ``inversion`` (no artificial duplication of some hypotheses referring to
+     other hypotheses). To take advantage of these improvements, it is enough to use
+     ``inversion … as []``, letting Coq choose fresh names.
+
 .. note::
    If :n:`@ident` does not denote a hypothesis in the local context but
    refers to a hypothesis quantified in the goal, then the latter is
    first introduced in the local context using :n:`intros until @ident`.
 
 .. note::
-   As ``inversion`` proofs may be large in size, we recommend the
-   user to stock the lemmas whenever the same instance needs to be
+   As ``inversion`` proofs may be large, we recommend
+   creating and using lemmas whenever the same instance needs to be
    inverted several times. See :ref:`derive-inversion`.
 
+.. todo PR: I find the language "if they relate two proofs ..." confusing.
+   Is there a point using "proof" here or can it be reworded to only
+   refer to types?  I've seen this idiom elsewhere in the chapter.
```

Comment:
What are denoted by "proofs" in this chapter is terms whose type is a proposition (i.e. has type `Prop`). So no, we cannot reword like you suggest.

Reply:
IIUC what you suggest, absolutely not! Proofs are terms whose type is a proposition and propositions are types whose sort is `Prop`. So the two are not at the same level.